src/ucp/core/ucp_ep.c

```diff
@@ -1034,16 +1032,60 @@ void ucp_ep_destroy(ucp_ep_h ep)
     return;
 }
 
-int ucp_ep_config_lane_is_equal(const ucp_ep_config_key_t *key1,
-                                const ucp_ep_config_key_t *key2,
-                                ucp_lane_index_t lane, int compare_types)
+static int
+ucp_ep_config_check_same_proxies(const ucp_ep_config_key_t *key1,
+                                 ucp_lane_index_t lane1,
+                                 const ucp_ep_config_key_t *key2,
+                                 ucp_lane_index_t lane2)
+{
+    ucp_lane_index_t proxy_lane1 = key1->lanes[lane1].proxy_lane;
+    ucp_lane_index_t proxy_lane2 = key2->lanes[lane2].proxy_lane;
+
+    return (/* both lanes are proxies for themselves */
+            (proxy_lane1 == lane1) && (proxy_lane2 == lane2))   ||
+           (/* both lanes are proxies for the same peer */
+            (proxy_lane1 != UCP_NULL_RESOURCE) &&
+            (proxy_lane2 != UCP_NULL_RESOURCE) &&
+            ucp_ep_config_lane_is_same_peer(key1, proxy_lane1,
+                                            key2, proxy_lane2)) ||
+           (/* both lanes are not proxies */
+            (proxy_lane1 == UCP_NULL_RESOURCE) &&
+            (proxy_lane2 == UCP_NULL_RESOURCE));
+}
+
+static int
+ucp_ep_config_lane_is_same_dst_rsc_index(const ucp_ep_config_key_t *key1,
+                                         ucp_lane_index_t lane1,
+                                         const ucp_ep_config_key_t *key2,
+                                         ucp_lane_index_t lane2)
+{
+    return (/* compare resource indices, if both are specified */
+            (key1->lanes[lane1].dst_rsc_index != UCP_NULL_RESOURCE) ==
+            (key2->lanes[lane2].dst_rsc_index != UCP_NULL_RESOURCE)) &&
+           (key1->lanes[lane1].dst_rsc_index  != key2->lanes[lane2].dst_rsc_index);
```

Comment:
why `!=`?

Reply:
fixed
it works since `ucp_ep_config_lane_is_same_dst_rsc_index()` was called without negation